gsa/src/web/components/powerfilter/severityvaluesgroup.js

```diff
@@ -16,64 +16,41 @@
  * along with this program; if not, write to the Free Software
  * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
  */
-import React from 'react';
+import React, {useState} from 'react';
 
 import PropTypes from 'web/utils/proptypes.js';
 
 import FormGroup from 'web/components/form/formgroup.js';
 import {parseSeverity} from 'gmp/parser.js';
 import RelationSelector from 'web/components/powerfilter/relationselector';
 import NumberField from 'web/components/form/numberfield';
-import Divider from '../layout/divider';
-
-class SeverityValuesGroup extends React.Component {
-  constructor(props) {
-    super(props);
-
-    this.state = {
-      relation: '=',
-    };
-
-    this.handleRelationChange = this.handleRelationChange.bind(this);
-  }
-
-  handleRelationChange(rel) {
-    this.setState({
-      relation: rel,
-    });
-    const severity = parseSeverity(this.props.filter.get(this.props.name));
-    const keyword = this.props.name;
-
-    this.props.onChange(severity, keyword, rel);
-  }
-
-  render() {
-    const severity = parseSeverity(this.props.filter.get(this.props.name));
-    const newRelation = this.state.relation;
-    const keyword = this.props.name;
-    return (
-      <FormGroup title={this.props.title}>
-        <Divider>
-          <RelationSelector
-            relation={newRelation}
-            onChange={this.handleRelationChange}
-          />
-          <NumberField
-            name={this.props.name}
-            type="int"
-            min={0}
-            max={10}
-            value={severity}
-            size="5"
-            onChange={(value = severity, name = keyword) =>
-              this.props.onChange(value, name, newRelation)
-            }
-          />
-        </Divider>
-      </FormGroup>
-    );
-  }
-}
+import Divider from 'web/components/layout/divider';
+
+const SeverityValuesGroup = ({filter, name, title, onChange}) => {
+  const [rel, setRel] = useState('=');
```

Comment:
I am thinking about another issue which has been in the component before already. Could it be possible that the relation is not "=" when initially being rendered. I guess we have to get the relation from the filter.

```js
const term = filter.getTerm(name);
severity = isDefined(term) ? parseSeverity(term.value) : undefined;
relation = isDefined(term) ? term.relation : '=';
...
```

Reply:
Thanks for this catch. I put the relation bit into the useState argument.